src/msg/async/AsyncConnection.h

```diff
@@ -104,8 +104,16 @@ class AsyncConnection : public Connection {
     return !out_q.empty() || outcoming_bl.length();
   }
   void shutdown_socket() {
-    if (sd >= 0)
+    for (auto &&t : register_time_events)
+      center->delete_time_event(t);
```

Comment:
and strictly speaking this changes the behaviour of `shutdown_socket()`. which has callers besides `AsyncConnection::fault()`. while the commit message implies this change is but a refactor. so this might fix some issue also?